src/Package/Impl/ProjectSystem/Commands/CopyRemoteItemPathCommand.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Microsoft.Common.Core.Shell;
+using Microsoft.R.Components.Extensions;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.R.Host.Client.Session;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.FileSystemMirroring;
+using Microsoft.VisualStudio.R.Package.Commands;
+using Microsoft.VisualStudio.R.Package.Shell;
+using static System.FormattableString;
+#if VS14
+using Microsoft.VisualStudio.ProjectSystem.Designers;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+#endif
+
+namespace Microsoft.VisualStudio.R.Package.ProjectSystem.Commands {
+    [ExportCommandGroup("AD87578C-B324-44DC-A12A-B01A6ED5C6E3")]
+    [AppliesTo(ProjectConstants.RtvsProjectCapability)]
+    internal sealed class CopyRemoteItemPathCommand : IAsyncCommandGroupHandler {
+        private readonly ConfiguredProject _configuredProject;
+        private readonly IRInteractiveWorkflowProvider _interactiveWorkflowProvider;
+        private readonly ICoreShell _coreShell;
+
+        [ImportingConstructor]
+        public CopyRemoteItemPathCommand(ConfiguredProject configuredProject, IRInteractiveWorkflowProvider interactiveWorkflowProvider, ICoreShell coreShell) {
+            _configuredProject = configuredProject;
+            _interactiveWorkflowProvider = interactiveWorkflowProvider;
+            _coreShell = coreShell;
+        }
+
+        public Task<CommandStatusResult> GetCommandStatusAsync(IImmutableSet<IProjectTree> nodes, long commandId, bool focused, string commandText, CommandStatus progressiveStatus) {
+            if (commandId == RPackageCommandId.icmdCopyRemoteItemPath && nodes.IsSingleNodePath() && _interactiveWorkflowProvider.GetOrCreate().RSession.IsRemote) {
```

Comment:
This may actually try and create the session just for status check. Maybe just IsHostRunning first?